### PR TITLE
Feature/improvements

### PR DESCRIPTION
**Pull request by VladMartyshevskyi**
1) Added checks for permissions specified in the editor config
2) Moved breadcrumb properties to the conf
3) Rerun WOPI Discovery in case of proof key validation failure, or success with the old key.
4) Refactoring

### CODE REVIEW

File: services/src/main/java/org/exoplatform/officeonline/WOPIService.java [hunk 1]
```diff
@@ -60,106 +62,110 @@
 import org.exoplatform.services.wcm.core.NodetypeConstant;
 import org.exoplatform.services.wcm.utils.WCMCoreUtils;
 
+// TODO: Auto-generated Javadoc
```
Comment (pnedonosko) [inline services/src/main/java/org/exoplatform/officeonline/WOPIService.java]: Remove it

File: services/src/main/java/org/exoplatform/officeonline/WOPIService.java [hunk 2]
```diff
@@ -487,14 +506,20 @@ public String renameFile(String fileId, EditorConfig config, String newTitle, St
...
-   * Checks if current user can rename the document.
+   * Checks parent node permissions.
    *
    * @param node the node
    * @return true if user can rename
    * @throws RepositoryException the repository exception
    */
-  protected boolean canRenameDocument(Node node) throws RepositoryException {
-    NodeImpl parent = (NodeImpl) node.getParent();
+  protected boolean hasParentPermissions(Node node) throws RepositoryException {
```
Comment (pnedonosko) [inline services/src/main/java/org/exoplatform/officeonline/WOPIService.java]: A better name would be `canModifyDocument()`

File: services/src/main/java/org/exoplatform/officeonline/rest/WOPIResource.java [hunk 3]
```diff
@@ -633,6 +669,11 @@ private Response putRelativeFile(EditorConfig config, String target, boolean ove
                      .entity("{\"error\": \"" + e.getMessage() + "\"}")
                      .type(MediaType.APPLICATION_JSON)
                      .build();
+    } catch (PermissionDeniedException e) {
+      return Response.status(Status.FORBIDDEN)
+                     .entity("{\"error\": \"" + e.getMessage() + "\"}")
```
Comment (pnedonosko) [inline services/src/main/java/org/exoplatform/officeonline/rest/WOPIResource.java:677]: Don't return direct exception message here (if it's only not ours one, already prepared for client/user side) but tells something clear an short and then use the same text in the LOG error/warn message. Apply to all such exception handlers.

File: webapp/src/main/webapp/WEB-INF/conf/officeonline/configuration.xml [hunk 4]
```diff
@@ -41,6 +41,10 @@
         <name>token-configuration</name>
         <property name="secret-key" value="${officeonline.token.secret:}" />
       </properties-param>
+      <properties-param>
+        <name>breadcrumb-configuration</name>
+        <property name="brand-name" value="${officeonline.brand.name:ExoPlatform}" />
```
Comment (pnedonosko) [inline webapp/src/main/webapp/WEB-INF/conf/officeonline/configuration.xml]: eXo Platform

**Review by pnedonosko — CHANGES_REQUESTED** [review]

File: webapp/src/main/webapp/WEB-INF/conf/officeonline/configuration.xml (hunk 4 above)
Comment (VladMartyshevskyi) [reply webapp/src/main/webapp/WEB-INF/conf/officeonline/configuration.xml]: Fixed: https://github.com/exo-addons/office-online/pull/10/commits/36cfb2d370ee05b1aab1b09e471285054e5971e8

File: services/src/main/java/org/exoplatform/officeonline/rest/WOPIResource.java (hunk 3 above)
Comment (VladMartyshevskyi) [reply services/src/main/java/org/exoplatform/officeonline/rest/WOPIResource.java:677]: Fixed: https://github.com/exo-addons/office-online/pull/10/commits/36cfb2d370ee05b1aab1b09e471285054e5971e8

File: services/src/main/java/org/exoplatform/officeonline/WOPIService.java (hunk 2 above)
Comment (VladMartyshevskyi) [reply services/src/main/java/org/exoplatform/officeonline/WOPIService.java]: Fixed: https://github.com/exo-addons/office-online/pull/10/commits/36cfb2d370ee05b1aab1b09e471285054e5971e8

File: services/src/main/java/org/exoplatform/officeonline/WOPIService.java (hunk 1 above)
Comment (VladMartyshevskyi) [reply services/src/main/java/org/exoplatform/officeonline/WOPIService.java]: Fixed: https://github.com/exo-addons/office-online/pull/10/commits/36cfb2d370ee05b1aab1b09e471285054e5971e8

**Review by pnedonosko — APPROVED** [review]

**Pull request merged**